azext_iot/operations/hub.py

```diff
@@ -1618,6 +1618,44 @@ def iot_hub_configuration_delete(
         handle_service_exception(e)
 
 
+def iot_hub_configuration_test_queries(
+    cmd,
+    target_condition=None,
+    custom_metric_queries=None,
+    hub_name=None,
+    resource_group_name=None,
+    login=None,
+    auth_type_dataplane=None
+):
+    discovery = IotHubDiscovery(cmd)
+    target = discovery.get_target(
+        resource_name=hub_name,
+        resource_group_name=resource_group_name,
+        login=login,
+        auth_type=auth_type_dataplane,
+    )
+    resolver = SdkResolver(target=target)
+    service_sdk = resolver.get_sdk(SdkType.service_sdk)
+
+    try:
+        if custom_metric_queries:
+            custom_metric_queries = process_json_arg(custom_metric_queries, argument_name="custom_metric_queries")
+
+        result = service_sdk.configuration.test_queries(target_condition, custom_metric_queries, raw=True).response.json()
+        validation_errors = []
+
+        if not result['targetConditionError'] and not result['customMetricQueryErrors']:
+            return 'Validation passed!'
+        if result['targetConditionError']:
+            validation_errors.append('Target condition validation failed: {}.'.format(result['targetConditionError']))
+        if result['customMetricQueryErrors']:
```

Comment:
Is there something non-optimal or confusing in regards to the data model returned by the service where we need to organize it in a different way?

Reply:
do you mean why I chose to use raw=True? yeah that's because when I try to mock response, Vicky and I tried different formats to pass in target_condition_error and custom_metric_query_errors, but all of them goes to 'additional properties' instead of the expected place. And using raw=True as Vicky suggested did fix this issue since the fixture now don't do assumption of where to put the properties that we mock anymore. Please let me know if there is better insight to fix this, thanks!😀